### PR TITLE
[#1.6]Added quit option and test

### PR DESCRIPTION
**Pull request by euloong**
- Updated showOptions with quit option.
- Updated manageOptions with System.exit(0) when quit option selected.
- Added test to check app is closed when quit option is selected, needed to install the System Rules library via the Maven interface.
- Forgot to pull from master when creating my new branch, will have to manually merge.

### CODE REVIEW

File: src/com/twu/biblioteca/MainMenu.java [hunk 1]
```diff
@@ -9,7 +9,9 @@ public void welcome() {
     }
 
     public void showOptions() {
-        System.out.println("Please select an option from the following:\n" + "1. List of books");
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:12]: I like this natural refactor to separate print lines once a second option was needed

File: src/com/twu/biblioteca/MainMenu.java [hunk 2]
```diff
@@ -31,6 +33,10 @@ public void manageOptions() {
                 showList();
                 counter = 1;
             }
+            else if(input.equals("2")) {
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:36]: Out of curiosity, what do you think would happen if the user entered " 2" or "2 " or something like that?  Is that something you could/should handle gracefully?

File: test/com/twu/biblioteca/MainMenuTest.java [hunk 3]
```diff
@@ -74,4 +81,17 @@ public void shouldDisplayInvalidMessageWhenInvalidOptionSelected() {
...
+
+    @Test
+    public void shouldCloseApplicationWhenQuitOptionSelected() {
+        MainMenu mainMenu = new MainMenu();
+
+        System.setIn(new ByteArrayInputStream(("2").getBytes()));
+
+        mainMenu.showOptions();
+
+        exit.expectSystemExit();
```
Comment (andeemarks) [inline test/com/twu/biblioteca/MainMenuTest.java:93]: Wow! This is a new on to me... I didn't know this was possible.  I suspect what this is doing is mocking out the System class and intercepting the call to "exit".

File: src/com/twu/biblioteca/MainMenu.java (hunk 2 above)
Comment (euloong) [reply src/com/twu/biblioteca/MainMenu.java:36]: Currently the code accepts it but it was not planned by me and I should have accounted for that by cleaning up the user input. I will add a trim method to clean up the user input.

File: test/com/twu/biblioteca/MainMenuTest.java (hunk 3 above)
Comment (euloong) [reply test/com/twu/biblioteca/MainMenuTest.java:93]: Yes, I had to use a 3rd party library to implement a rule for SystemExit.

https://stefanbirkner.github.io/system-rules/index.html

It was the only way I could out to test the quit option though I suspect there must be other out of the box ways to do it.

Comment (andeemarks) [reply test/com/twu/biblioteca/MainMenuTest.java:93]: The only other way I know of is doing a system-level test where you start the application in the test (e.g., "java ..."), feed it input and then test that it has exited.  But that approach would be beyond the scope of this work and something I wouldn't normally test in most cases

**Pull request merged**